### DISCUSSION TITLE
API changes / add

### DISCUSSION THREAD

**Discussion by TheBeast2155** [pterodactyl/panel#3192] [Feature Requests]
When you update a server via the API, why do you need to firstly, get the server build details, and then send it back again if you only want to update one part of it. If I just want to update the memory to the server, must send swap, io and so one to update just the memory. It would be much smarter if it’s not set in the API call, it would just ignore it and update the items that are populated. 

It would also be very nice to have a API that returns a unused allocation, without looping through the nodes API list, and then the allocation API after, to find one that is not in use. 

And just for peace of mind, make it possible to change the API call limit yourself. It’s my own API, and it’s my server resources. Let me decide the limit.

**Comment by DaneEveritt** [comment]
You can already control the API throttles with environment variables: https://github.com/pterodactyl/panel/blob/develop/config/http.php

As for the server updating, that is just how the API works.
